### DISCUSSION TITLE
Best way to Wrap or Extend HxInputText for Validation?

### DISCUSSION THREAD

**Discussion by StarGazer202424** [havit/Havit.Blazor#807] [Q&A]
Hi

None of the validations that come out of the box for input validation works for my use case. I am stuck with using the custom validation, but the validation I need to use needs to be on every control like

```
<label>
 My Label
<HxValidationMessage For="() => model.Text />
<HxInputText @bind-Value="@model.Text"  ValidationMessageMode="ValidationMessageMode.None" />
</label>
```

As you can see, I want the Validation Message to be above the Input Text instead of under it. 

If I need to do 10 of these, that will be a lot of copying and pasting. I was thinking of wrapping it into a component, but I am not sure how to make it, so I can still pass in all the parameters that can be set on an HxInputText

**Comment by hakenr** [comment]
When dealing with such requirements, you'll need to implement your own [`FormValueRenderer`](https://github.com/havit/Havit.Blazor/blob/master/Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/FormValueRenderer.cs) and set it to `FormValueRenderer.Current` (a static field that provides an ambient context for all our form component renderings).

Check out the default implementation of rendering at https://github.com/havit/Havit.Blazor/blob/master/Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxFormValueRenderer.cs and https://github.com/havit/Havit.Blazor/blob/master/Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxFormValueComponentRenderer.cs.

**Reply by StarGazer202424** [reply]
Great let me check that out and get back to you if I got questions

**Comment by hakenr** [comment]
(We can implement such renderer for you as part of our commercial support. Let me know if you are interested in such option.)

**Reply by StarGazer202424** [reply]
It's a bit hard to get approval for this kind of stuff but how much is commercial support?

**Reply by hakenr** [reply]
Sorry for the delayed response. We are currently shaping our commercial support offer. In the meantime, any one-off sponsorship (via GitHub Sponsorship) higher than $150 will allow us to create the custom `FormValueRenderer` (and related files) for you.
